drltrace/scripts/gen_drlttrace_config.py

```diff
@@ -0,0 +1,379 @@
+
+def parse_prefix(line):
+    '''
+    The function searches for a WinAPI return type in the string specified
+    @in line - a string with a type to find
+    @out - returning type
+    '''
+    if "//" in line or "Boolean" in line or "&&" in line:
+        return None
+    if ")" in line:
+        line = parse_internal_parentesses(line)
+
+    line = line.split(" ")
+    func_type = None
+    has_type = 0
+    ignore_until_paren = False
+    for element in line:
+        if "__success" in element:
+            ignore_until_paren = True
+        if ")" in element:
+            ignore_until_paren = False
+            continue
+        if ignore_until_paren == True:
+            continue
+        if element in tokens_ignore_prefix or element == "" or "__" in element\
+           or (("WIN" in element or "NT" in element) and "API" in element):
+            continue
+        if has_type >= 1:
+            print "Warning. Failed to parse line %s (%s)" % (line, element)
+            return "None"
+        func_type = element
+        has_type += 1
+
+    if func_type != None:
+        save_type = func_type
+        func_type = get_type(func_type)
+        if func_type == "None":
+            try:
+                unknown_types[save_type] += 1
+            except:
+                unknown_types[save_type] = 1
+
+    return func_type
+
+def parse_args(args_str):
+    if "//" in args_str:
+        return "FAILED"
+    args_str = args_str.replace("\n", "")
+    args_str = args_str.replace(");", "")
+    if ")" in args_str:
+        args_str = parse_internal_parentesses(args_str)
+    args_str = args_str.replace("(", "")
+    args = args_str.split(",")
+    #if len(args) <= 1:
+        #return "FAILED"
+    res_types = list()
+    for arg in args:
+        arg = arg.split(" ")
+        param_type = 0
+        pointer = 0
+        res_type = None
+        for element in arg:
+            if "__out" in element or "__deref_out" in element or "__deref_opt_out_opt" in element or "__deref_opt_out" in element:
+                param_type = 1
+                continue
+            elif "__inout" in element:
+                param_type = 2
+                continue
+            if "*" in element:
+                pointer = 1
+                element = element.replace("*", "")
+            type = analyze_type(element)
+            if type != None:
+                res_type = type
+            #else:
+                #try:
+                    #unknown_types[element] += 1
+                #except:
+                    #unknown_types[element] = 1
+        if res_type == None:
+            print "Unknown type for us"
+        else:
+            if param_type == 1:
+                res_type = "+" + res_type
+            elif param_type ==  2:
+                res_type = "++" + res_type
+            if pointer == 1:
+                res_type += "*"
+        res_types.append(str(res_type))
+    return "|".join(res_types)
+
+def parse_args(args_str):
```

Comment:
Sorry, this is the first time I'm commiting Python code in the DrMemory repo. Next time I will review it more carefully.